src/App/Extensions/SDKExtensions.cs

```diff
@@ -177,6 +168,35 @@ private static async Task Connect(ISDK sdk, ILocation SpecificLocation = null, b
             }
         }
 
+        private async static Task Connect(ISDK sdk, ILocation location, List<IConnectionConfiguration> configurations, CancellationToken token, List<ILocation> locations = null)
+        {
+            // If the network connection type is Unspecified, this means that the Automatic Protocol feature is selected.
+            if (Properties.Settings.Default.ConnectionProtocol == NetworkConnectionType.Automatic)
+            {
+                if (locations == null)
+                {
+                    await sdk.Connect(location, configurations, token); 
+                }
+                else 
+                { 
+                    await sdk.Connect(locations, configurations, token); 
+                }
```

Comment:
Simplify this using ?? null-coalescing operator

`await sdk.Connect(locations ?? location, configurations, token);`

Reply:
Added null-coalescing operator